### PR TITLE
Add __hash__ for IntEnum

### PR DESCRIPTION
**Pull request by HPLegion**
Hi  :-)

Partially solving #5911

This patch adds `__hash__` capability for IntEnums. I think this would be a nice addition, because I like to use enums as keys for the typed `Dict`, since there is less risk of introducing typos. 

I have been using this inside my own code for a while, but I had mistakenly thought that the hashes I created are different from the CPython hashes.
Today I realised that the CPython hash for an IntEnum is in fact simply `int.__hash__ (value)` so I think this could move into numba.

Is it frowned upon to use the extension API for internals? 

Cheers!

### CODE REVIEW

**Comment by stuartarchibald** [comment]
Thanks for the patch @HPLegion 

> Is it frowned upon to use the extension API for internals?

It's encouraged!

File: numba/cpython/enumimpl.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
                                  lower_getattr_generic, lower_cast,
                                  lower_constant, impl_ret_untracked)
 from numba.core import types
+from numba.extending import overload_method
```
Comment (stuartarchibald) [inline numba/cpython/enumimpl.py:10]:
```suggestion
from numba.core.extending import overload_method
```

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch, one minor thing to resolve else looks good.

File: numba/cpython/enumimpl.py (hunk 1 above)
Comment (HPLegion) [reply numba/cpython/enumimpl.py]: Thanks for the suggestions. I pushed an update :-)

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch and fixes.

**Pull request merged**